### PR TITLE
Added .asize and .isize pseudo variables in ca65

### PR DESCRIPTION
**Pull request by undisbeliever**
These pseudo variables will return the size of the accumulator/index
in bits.

For the 65816 instruction set .ASIZE/.ISIZE will return either 8 or 16,
depending on the current size of the operand in immediate addressing
mode.

For all other CPU instruction sets, .ASIZE/.ISIZE will always return 8.

For example:

	; Reverse Subtract with Accumulator
	; A = memory - A
	.macro rsb param
		.if .asize = 8
			eor	#$ff
		.else
			eor	#$ffff
		.endif
		sec
		adc	param
	.endmacro

### CODE REVIEW

File: src/ca65/expr.c [hunk 1]
```diff
@@ -1099,6 +1099,15 @@ static ExprNode* Factor (void)
             N = Function (FuncAddrSize);
             break;
 
+	case TOK_ASIZE:
+            if (GetCPU() != CPU_65816) {
+                N = GenLiteralExpr (8);
+            } else {
+	        N = GenLiteralExpr (ExtBytes [AM65I_IMM_ACCU] * 8);
```
Comment (oliverschmidt) [inline src/ca65/expr.c]: Just to prove my nitpicking: Here one space is missing ;-)

Comment (greg-king5) [inline src/ca65/expr.c]: I suspect that Oliver was thinking about two lines at the same time -- he got his lines crossed!  ;-)

His comment is for the third line above this one: `GetCPU ()`.

This line needs a space _removed_: `ExtBytes[AM65I_IMM_ACCU]`.

**Comment by greg-king5** [comment]
Except for the nits, this Pull Request is good.

And, a macro definitely needs to be able to learn which size ca65's `.smart` feature has chosen.

**Comment by oliverschmidt** [comment]
@greg-king5: Thanks for your assessment :-)

**Comment by oliverschmidt** [comment]
@undisbeliever: First of all thanks for your contribution to cc65 :-)

I've noticed that you wanted to make sure that your code (incl. doc) fits to rest style-wise. So I presume that you want to complete this by "fixing" the space.

If yes: Please remember to write a short comment here as I don't get notifications for new commits to existing pull request branches.

If no: State so here. I'm willing to merge this pull request as-is and "fix" the space myself. Note: Regular contributors are generally required to finalize the content themselves.

**Comment by undisbeliever** [comment]
@oliverschmidt 

Changes to the code style have been made.

**Comment by oliverschmidt** [comment]
> Changes to the code style have been made.

:-)

**Pull request merged**